layouts/partials/blocks/candidate-results.html

```diff
@@ -0,0 +1,80 @@
+{{ with .block }}
+  {{ $data := index site.Data.candidates2024 }}
+
+  <script>
+    function candidates() {
+
+      let urlParams = new URL(window.location).searchParams.entries();
+      let candidateData = {{$data}}
```

Comment:
This is probably I don't like mixing JS and template data because it gets hairy fast, so I would probably do something like 

```html
<script data-candidates="{{ $data }}">
    function candidates() {
      let candidateData = JSON.parse(document.currentScript.dataset.candidates);
```

Probably not worth changing though because it's a one off.